lib/view_component/base.rb

```diff
@@ -83,6 +83,18 @@ def self._deprecated_generate_mattr_accessor(name)
     _deprecated_generate_mattr_accessor :sidecar
     _deprecated_generate_mattr_accessor :stimulus_controller
 
+    # In order for translations to work in an initializer,
+    # ActionView requires @virtual_path to be set. To set it,
+    # we allocate the component, set @virtual_path, then
+    # call the component's initializer. This is either terrible,
+    # brilliant, or both.
+    def self.new(...)
+      precursor = allocate
+      precursor.instance_variable_set(:@virtual_path, virtual_path)
+      precursor.send(:initialize, ...)
```

Comment:
In general I don't think the indirection is worth it when it comes to `self.new`, but setting that aside I don't think we should move forward with this if this is only fixing translations in the initializer, since that might set the expectation that *all* behavior should be runnable in the initializer.

I'd love to see use-cases where translations are necessary in the initializer since I imagine a better pattern would be to push folks towards using some other hook (e.g. `before_render`) instead of relying on the initializer to setup view data. That would also help co-locate logic that needs to run before a component is rendered.